orangecontrib/text/widgets/ownyt.py

```diff
@@ -132,42 +132,48 @@ def __init__(self):
                                         focusPolicy=Qt.NoFocus)
 
     def new_query_input(self):
-        if self.running:
-            self.stop()
+        self.search.stop()
         self.search()
 
     def start_stop(self):
-        if self.running:
-            self.stop()
+        if self.search.running:
+            self.search.stop()
         else:
             self.query_box.synchronize(silent=True)
             self.search()
 
     @gui_require('nyt_api', 'no_api')
     @gui_require('recent_queries', 'no_query')
-    @asynchronous(allow_partial_results=True)
-    def search(self, on_progress, should_break):
+    def run_search(self):
+        self.search()
+
+    @asynchronous
+    def search(self):
+        return self.nyt_api.search(self.recent_queries[0], self.date_from, self.date_to,
+                                   on_progress=self.progress_with_info,
+                                   should_break=self.search.should_break)
+
+    @search.callback
+    def progress_with_info(self, n_retrieved, n_all):
         self.Error.api_error.clear()
```

Comment:
Why is this error clearing is moved from `search` to `callback`? Won't it happen later after this change?